clang/include/clang/Basic/OpenACCKinds.h

```diff
@@ -0,0 +1,72 @@
+//===--- OpenACCKinds.h - OpenACC Enums -------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Defines some OpenACC-specific enums and functions.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_BASIC_OPENACCKINDS_H
+#define LLVM_CLANG_BASIC_OPENACCKINDS_H
+
+namespace clang {
+// Represents the Construct/Directive kind of a pragma directive. Note the
+// OpenACC standard is inconsistent between calling these Construct vs
+// Directive, but we're calling it a Directive to be consistent with OpenMP.
+enum class OpenACCDirectiveKind {
+  // Compute Constructs.
+  Parallel,
```

Comment:
I would suggest starting with a single construct support, say parallel. Other constructs must be added later, when each particular feature is going to be implemented.

Reply:
Our implementation strategy is to start with all of the 'parsing' first, starting at constructs, I don't see why this is an issue?  I understand the grammar enough that this seems valid.  

At this point, we HAVE successfully implemented parsing for all of the 'non-optional-parens' versions.  

Doing it this way permits us to do the implementation of clauses next: clauses in OpenACC are shared between all of the constructs, so implementing 1 construct at a time would prevent us from being able to do that.

Reply:
To clarify: Are you asking me to split this patch up into 23 different ones here?  The first adding the infrastructure + parallel, followed by most of the 22 that are just adding an entry here and to a string-switch?

Reply:
No, not neccesary. The first patch, that lands main infrastructure, better to have a small parsing functionality, the future patch(es) may include parsing of other stuff, if it small enough

Reply:
Alright then, I'll do that.